httpclient5-fluent/src/main/java/org/apache/hc/client5/http/fluent/ContentResponseHandler.java

```diff
@@ -44,11 +46,63 @@
  */
 public class ContentResponseHandler extends AbstractHttpClientResponseHandler<Content> {
 
+    /**
+     * Handles the response entity and transforms it into a {@link Content} object. If the response entity is null or
+     * has zero length, {@link Content#NO_CONTENT} is returned.
+     *
+     * @param entity the response entity.
+     * @return a {@link Content} object that encapsulates the response body, or {@link Content#NO_CONTENT} if the
+     * response body is null or has zero length.
+     * @throws IOException if an I/O error occurs.
+     */
     @Override
     public Content handleEntity(final HttpEntity entity) throws IOException {
-        return entity != null ?
-                new Content(EntityUtils.toByteArray(entity), ContentType.parse(entity.getContentType())) :
-                Content.NO_CONTENT;
+        if (entity != null && entity.getContentLength() > 0) {
+            return new Content(EntityUtils.toByteArray(entity), ContentType.parse(entity.getContentType()));
+        } else {
+            return Content.NO_CONTENT;
+        }
+    }
+
+    /**
+     * Handles a successful response (2xx status code) and returns the response entity as a {@link Content} object.
+     * If no response entity exists, {@link Content#NO_CONTENT} is returned.
+     *
+     * @param response the HTTP response.
+     * @return a {@link Content} object that encapsulates the response body, or {@link Content#NO_CONTENT} if the
+     * response body is null or has zero length.
+     * @throws HttpResponseException if the response was unsuccessful (a >= 300 status code).
+     * @throws IOException           if an I/O error occurs.
+     */
+    @Override
+    public Content handleResponse(final ClassicHttpResponse response) throws IOException {
+        final int statusCode = response.getCode();
+        final HttpEntity entity = response.getEntity();
+        final Content content = handleEntity(entity);
+        if (statusCode >= 300) {
+            final StringBuilder messageBuilder = new StringBuilder(response.getReasonPhrase());
+            if (content != null && content.getType() != null) {
+                messageBuilder.append(", body was ");
+                truncateString(content.asString(), messageBuilder);
```

Comment:
@arturobernalg The truncation logic looks flawed. What is the point of truncating the message if it has already been fully loaded in memory? Use `EntityUtils#toByteArray` with the content max limit instead.